src/app/components/operations/append.js

```diff
@@ -14,16 +14,28 @@ class Append extends Component {
   handleSubmit = () => {
     const { makeRequest, form: { path = '', user = '' } } = this.props;
 
+    const fd = new FormData();
+    fd.append('file', this.state.file);
     makeRequest({
       endpoint: `/webhdfs/v1${path}`,
       method: 'POST',
+      body: fd,
       qs: {
         op: 'APPEND',
         'user.name': user,
       },
     });
   };
 
+  handleFileUpload = (event) => {
+    const fileReader = new FileReader();
+    const raw = event.target.files[0];
+    fileReader.readAsDataURL(raw);
```

Comment:
Why `readAsDataURL`? Wouldn't it be easier for the server if we read it as raw text?